iam_pipes.tf

```
@@ -0,0 +1,396 @@
+        "kinesis:GetShardIterator",
+        "kinesis:ListShards",
+        "kinesis:ListStreams",
+        "kinesis:SubscribeToShard"
+      ]
+    }
+
+    kinesis_target = {
+      actions = [
+        "kinesis:PutRecord"
+      ]
+    }
+
+    mq = {
+      actions = [
+        "mq:DescribeBroker",
+        "secretsmanager:GetSecretValue",
+        "ec2:CreateNetworkInterface",
+        "ec2:DeleteNetworkInterface",
+        "ec2:DescribeNetworkInterfaces",
+        "ec2:DescribeSecurityGroups",
+        "ec2:DescribeSubnets",
+        "ec2:DescribeVpcs",
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+    }
+
+    msk = {
+      # Read this for more: https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-pipes-msk.html#pipes-msk-permissions-iam-policy
+      actions = [
+        "kafka:DescribeClusterV2",
+        "kafka:GetBootstrapBrokers",
+        "ec2:CreateNetworkInterface",
+        "ec2:DeleteNetworkInterface",
+        "ec2:DescribeNetworkInterfaces",
+        "ec2:DescribeSecurityGroups",
+        "ec2:DescribeSubnets",
+        "ec2:DescribeVpcs",
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+    }
+
+    lambda = {
+      actions = [
+        "lambda:InvokeFunction"
+      ]
+    }
+
+    step_functions = {
+      actions = [
+        "states:StartExecution"
+      ]
+    }
+
+    api_gateway = {
+      actions = [
+        "execute-api:Invoke"
+      ]
+    }
+
+    api_destination = {
+      actions = [
+        "events:InvokeApiDestination"
+      ]
+    }
+
+    batch = {
+      actions = [
+        "batch:SubmitJob"
+      ]
+    }
+
+    logs = {
+      actions = [
+        "logs:DescribeLogGroups",
+        "logs:DescribeLogStreams",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+    }
+
+    ecs = {
+      actions = [
+        "ecs:RunTask"
+      ]
+    }
+
+    ecs_iam_passrole = {
+      actions = [
+        "iam:PassRole"
+      ]
+    }
+
+    eventbridge = {
+      actions = [
+        "events:PutEvents"
+      ]
+    }
+
+    firehose = {
+      actions = [
+        "firehose:PutRecord"
+      ]
+    }
+
+    inspector = {
+      actions = [
+        "inspector:CreateAssessmentTemplate" # ???
+      ]
+    }
+
+    redshift = {
+      actions = [
+        "redshift-data:ExecuteStatement" # ???
+      ]
+    }
+
+    sagemaker = {
+      actions = [
+        "sagemaker:CreatePipeline" # ???
+      ]
+    }
+
+    sns = {
+      actions = [
+        "sns:Publish"
+      ]
+    }
+  }
+}
+
+#################################
+# IAM role for EventBridge Pipes
+#################################
+
+data "aws_iam_policy_document" "assume_role_pipe" {
+  for_each = local.role_for_pipes
+
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["pipes.amazonaws.com"]
```

We'll need to add `data "aws_partition" "current" {}`, but to support other non-commercial regions:

```suggestion
      identifiers = ["pipes.${data.aws_partition.current.dns_suffix}"]
```